### PR TITLE
Add Korrel8r deployment within COO

### PR DESCRIPTION
**Pull request by shwetaap**
Add Korrel8r deployment within observability operator

### CODE REVIEW

File: deploy/dependencies/korrel8r/korrel8r_svc.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v1
+kind: Service
+metadata:
```
Comment (alanconway) [inline deploy/dependencies/korrel8r/korrel8r_svc.yaml:3]: Need annotation to request serving certs from OCP ,see:
https://github.com/korrel8r/operator/blob/main/controllers/korrel8r_controller.go#L343

File: pkg/controllers/korrel8r/components.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			Selector: componentLabels(name),
+			Type:     corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func componentLabels(name string) map[string]string {
+	return map[string]string{
+		"app.kubernetes.io/instance":   name,
+		"app.kubernetes.io/part-of":    "Korrel8r",
```
Comment (alanconway) [inline pkg/controllers/korrel8r/components.go:119]: Need a consistent interpretation for these app.k8s.io labels.
E.g. is korrel8r part-of COO or part-of itself? Is korrel8r a component or does it have components?
Is instance a unique name within a namespace or unique cluster-wide?

The k8s docs are a bit ambiguous. I'm not so concerned about finding the "right" answer, more about being consistent with the rest of COO.

File: pkg/controllers/korrel8r/components.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+func korrel8rComponentReconcilers(korrel8rDeploy *appsv1.Deployment, korrel8rSvc *corev1.Service, korrel8rCfg Korrel8rConfiguration, namespace string) []reconciler.Reconciler {
+
+	return []reconciler.Reconciler{
+		reconciler.NewUpdater(newKorrel8rService(korrel8rObjName, namespace), korrel8rSvc),
+		reconciler.NewUpdater(newKorrel8rDeployment(korrel8rObjName, namespace, korrel8rCfg), korrel8rDeploy),
+	}
+}
+
+func newKorrel8rDeployment(name string, namespace string, korrel8rCfg Korrel8rConfiguration) *appsv1.Deployment {
+	deploy := &appsv1.Deployment{
```
Comment (alanconway) [inline pkg/controllers/korrel8r/components.go:31]: What's the relationship between this code and the YAML files above?

File: pkg/controllers/korrel8r/controller.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+	}
+
+	// korrel8rRoute := &routev1.Route{}
+	// if err := rm.k8sClient.Get(ctx, req.NamespacedName, korrel8rRoute); err != nil {
+	// 	if err != nil {
+	// 		// retry since some error has occured
+	// 		return ctrl.Result{}, err
+	// 	}
+	// }
+
```
Comment (alanconway) [inline pkg/controllers/korrel8r/controller.go:113]:
```suggestion
```
Not needed

**Review by alanconway — COMMENTED** [review]
Looks good. I was unsure why there are korrel8r_svc.yaml and korrel8r_deploy.yaml when it looks like this is (almost) what the controller code generates. Othewise it all makes sense.

File: cmd/operator/main.go [hunk 5]
```diff
@@ -39,6 +39,7 @@ var defaultImages = map[string]string{
 	"prometheus":    "",
 	"alertmanager":  "",
 	"thanos":        obopo.DefaultThanosImage,
+	"korrel8r": "quay.io/korrel8r/korrel8r:latest",
```
[the remark this answers is not in the document]
Comment (danielmellado) [reply cmd/operator/main.go:42]: +1 to that, we're also trying to promote this behavior later on for coo 0.3.0

File: pkg/controllers/korrel8r/controller.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+})
+
+// Korrel8rReconciler reconciles a Korrel8r object
+type resourceManager struct {
+	k8sClient  client.Client
+	scheme     *runtime.Scheme
+	logger     logr.Logger
+	controller controller.Controller
+	korrel8rconf Korrel8rConfiguration
+	version      string
```
Comment (danielmellado) [inline pkg/controllers/korrel8r/controller.go:44]: already marked by Github, but version is unused. I assume that as this is a draft it'll be filled later ;)

**Comment by shwetaap** [comment]
Closing this PR, and moving the work to here - https://github.com/rhobs/observability-operator/pull/497

**Pull request closed without merging**